pythonforandroid/recipes/python2legacy/__init__.py

```diff
@@ -0,0 +1,232 @@
+            if 'openssl' in self.ctx.recipe_build_order:
+                recipe = Recipe.get_recipe('openssl', self.ctx)
+                openssl_build_dir = recipe.get_build_dir(arch.arch)
+                setuplocal = join('Modules', 'Setup.local')
+                shprint(sh.cp, join(self.get_recipe_dir(), 'Setup.local-ssl'), setuplocal)
+                shprint(sh.sed, '-i.backup', 's#^SSL=.*#SSL={}#'.format(openssl_build_dir), setuplocal)
+                env['OPENSSL_VERSION'] = recipe.version
+
+            if 'sqlite3' in self.ctx.recipe_build_order:
+                # Include sqlite3 in python2legacy build
+                recipe = Recipe.get_recipe('sqlite3', self.ctx)
+                include = ' -I' + recipe.get_build_dir(arch.arch)
+                lib = ' -L' + recipe.get_lib_dir(arch) + ' -lsqlite3'
+                # Insert or append to env
+                flag = 'CPPFLAGS'
+                env[flag] = env[flag] + include if flag in env else include
+                flag = 'LDFLAGS'
+                env[flag] = env[flag] + lib if flag in env else lib
+
+            # NDK has langinfo.h but doesn't define nl_langinfo()
+            env['ac_cv_header_langinfo_h'] = 'no'
+            configure = sh.Command('./configure')
+            shprint(configure,
+                    '--host={}'.format(env['HOSTARCH']),
+                    '--build={}'.format(env['BUILDARCH']),
+                    # 'OPT={}'.format(env['OFLAG']),
+                    '--prefix={}'.format(realpath('./python-install')),
+                    '--enable-shared',
+                    '--disable-toolbox-glue',
+                    '--disable-framework',
+                    _env=env)
+
+            # tito left this comment in the original source. It's still true!
+            # FIXME, the first time, we got a error at:
+            # python$EXE ../../Tools/scripts/h2py.py -i '(u_long)' /usr/include/netinet/in.h  # noqa
+            # /home/tito/code/python-for-android/build/python/Python-2.7.2/python: 1: Syntax error: word unexpected (expecting ")")  # noqa
+            # because at this time, python is arm, not x86. even that,
+            # why /usr/include/netinet/in.h is used ?
+            # check if we can avoid this part.
+
+            make = sh.Command(env['MAKE'].split(' ')[0])
+            print('First install (expected to fail...')
+            try:
+                shprint(make, '-j5', 'install', 'HOSTPYTHON={}'.format(hostpython),
+                        'HOSTPGEN={}'.format(hostpgen),
+                        'CROSS_COMPILE_TARGET=yes',
+                        'INSTSONAME=libpython2.7.so',
+                        _env=env)
+            except sh.ErrorReturnCode_2:
+                print('First python2 make failed. This is expected, trying again.')
+
+            print('Second install (expected to work)')
+            shprint(sh.touch, 'python.exe', 'python')
+            shprint(make, '-j5', 'install', 'HOSTPYTHON={}'.format(hostpython),
```

Comment:
minor-out-of-scope-suggestion: we may want to use `cpu_count` from `multiprocessing` e.g. `'-j{}'.format(cpu_count())`

Reply:
Moving the request to a separate issue → https://github.com/kivy/python-for-android/issues/1558.